doc/classes/@GlobalScope.xml

```diff
@@ -2882,20 +2883,28 @@
 			[b]Note:[/b] The trailing colon is required for properly detecting built-in types.
 		</constant>
 		<constant name="PROPERTY_HINT_NODE_PATH_TO_EDITED_NODE" value="24" enum="PropertyHint">
+			Hints that a [NodePath] property should be relative to the scene root. The inspector editor will be initialized with [constant PROPERTY_USAGE_NODE_PATH_FROM_SCENE_ROOT].
```

Comment:
This hint is weird. It basically makes the property edited as if it was `PROPERTY_USAGE_NODE_PATH_FROM_SCENE_ROOT`, but it doesn't need to have this usage. Not sure what's it for.

Reply:
I don't think this description is correct. Both this hint and `PROPERTY_HINT_NODE_PATH_VALID_TYPES` initialize in a similar manner. The difference is that this hint sets the "base hint" type, and the other one sets the "valid types".

I'm not sure what you meant by "will be initialized with `PROPERTY_USAGE_NODE_PATH_FROM_SCENE_ROOT`", but we only check, in both cases, if `PROPERTY_USAGE_NODE_PATH_FROM_SCENE_ROOT` is set as a usage flag. And if it is, we set `use_path_from_scene_root` to true.

Then in `get_base_node()` we try to find the node _from the editor root_ based on the "base hint". We use it if we succeed (and the `use_path_from_scene_root` flag is then never used). If we don't succeed, the normal resolution logic kicks in, same as it would with the `PROPERTY_HINT_NODE_PATH_VALID_TYPES` hint.

This hint doesn't allow to set valid types at all, which means drag'n'drop doesn't work and the scene tree dialog doesn't do any filtering.

----

I agree that it's strange though. I would probably remove the code for this property and state in the docs that it's not used at this time. It seems to be a pretty old feature, maybe a part of some debugging at some ancient time.

Reply:
Should it get deprecated?

Reply:
I'd say so, since we have no use for it.

Reply:
The remaining two are still being assigned. So I guess that code should be removed.